src/components/Calendar/calendarHeader/single.js

```diff
@@ -0,0 +1,159 @@
+            },
+            [RIGHT_KEY]: () => {
+                const delta = disableNextMonth ? 2 : 1;
+                setFocusedItemIndex(Math.min(focusedItemIndex + delta, refs.length - 1));
+            },
+        }),
+        [disableNextMonth, disablePreviousMonth, focusedItemIndex, refs.length],
+    );
+
+    const handleKeyDown = useCallback(
+        event => {
+            if (disableNextMonth && disablePreviousMonth) return;
+            if (enableNavKeys) {
+                const { keyCode } = event;
+                if (keyHandlerMap[keyCode]) {
+                    event.preventDefault();
+                    event.stopPropagation();
+                    keyHandlerMap[keyCode]();
+                }
+            }
+        },
+        [enableNavKeys, keyHandlerMap, disableNextMonth, disablePreviousMonth],
+    );
+
+    const handleOnFocus = useCallback(index => {
+        setFocusedItemIndex(index);
+        setEnableNavKeys(true);
+    }, []);
+
+    const handleOnBlur = useCallback(() => {
+        setEnableNavKeys(false);
+    }, []);
+
+    useEffect(() => {
+        if (enableNavKeys && refs[focusedItemIndex].current) {
+            refs[focusedItemIndex].current.focus();
+        }
+    }, [enableNavKeys, focusedItemIndex, refs]);
+
+    return (
+        <StyledControlsContainer>
+            <StyledMonthContainer>
+                <StyledArrowButton
+                    ref={refs[0]}
+                    onClick={handlePrevMonthClick}
+                    size="medium"
+                    disabled={disablePreviousMonth}
+                    icon={<LeftIcon />}
+                    assistiveText="Previous Month"
+                    onKeyDown={handleKeyDown}
+                    onFocus={() => handleOnFocus(0)}
+                    onBlur={handleOnBlur}
+                />
+
+                <StyledMonth id={monthLabelId} data-id="month">
+                    {formattedMonth}
+                </StyledMonth>
+
+                <StyledArrowButton
+                    ref={refs[1]}
+                    onClick={handleNextMonthClick}
+                    size="medium"
+                    tabIndex={disablePreviousMonth ? undefined : -1}
+                    disabled={disableNextMonth}
+                    icon={<RightIcon />}
+                    assistiveText="Next Month"
+                    onKeyDown={handleKeyDown}
+                    onFocus={() => handleOnFocus(1)}
+                    onBlur={handleOnBlur}
+                />
+            </StyledMonthContainer>
+            <Select
+                ref={refs[2]}
+                label="select year"
+                hideLabel
+                tabIndex={disablePreviousMonth && disableNextMonth ? undefined : -1}
+                value={currentYear}
+                options={yearsRange}
+                onChange={onYearChange}
+                onClick={handleYearSelectClick}
+                onKeyDown={handleKeyDown}
```

Comment:
I think it is not necessary to change the Select API, you can put the onKeyDown = {handleKeyDown} in StyledContainer